### PR TITLE
Create anonymized columns method

### LINKED ISSUE #546: Add `create_anonymized_columns` method to anonymize data from scratch
### Problem Description

There are some transformers such as `AnonymizedFaker` and `RegexGenerator` that drop columns on the forward transform. It isn't possible to reverse transform _only_ these columns because there is nothing to input.

To allow for this case, we should add a new method: `create_anonymized_columns`

### Expected behavior

This method should belong to the `HyperTransformer` class and should be used only after fitting the data.

**Parameters:**
- (required) `num_rows`: An integer > 0 that describes the number of rows to anonymize
- (required) `column_names`: The list of column names to anonymize

```python
ht.fit(data)
ht.create_anonymized_columns(num_rows=10, column_names=['credit_card', 'user_id'])
```

### Error Cases

**HyperTransformer has not been fit yet.** The HyperTransformer must be fit in order to use this method.

```python
ht.create_anonymized_columns(num_rows=10, column_names=['credit_card', 'user_id'])
The HyperTransformer is not ready to use. Please fit your data first using 'fit' or 'fit_transform'.
```

**Required parameters: `num_rows` and `column_names`**. Throw an error if these aren't present or if there are extra, unknown parameters.

```python
ht.create_anonymized_columns(num_rows=10)
Error: Missing required parameter 'column_names'

ht.create_anonymized_columns(num_rows=10, column_names=['user_id'], test_parameter=4)
Error: Unknown parameter 'test_parameter'
```

**Parameter `num_rows`** must be an integer >0.

```python
ht.create_anonymized_columns(num_rows=-6, column_names=['user_id'])
Error: Parameter 'num_rows' must be an integer greater than 0.
```

**Parameter `column_names` must describe valid columns** that are in the config _and_ they must be assigned to `AnonymizedFaker` or `RegexGenerator` only.

```python
ht.create_anonymized_columns(num_rows=10, column_names=['invalid_name', 'credit_card'])
Error: Unknown column name 'invalid_name'. Use 'get_config()' to see a list of valid column names.

ht.create_anonymized_columns(num_rows=10, column_names=['age', 'credit_card'])
Error: Column 'age' cannot be anonymized. All columns must be assigned to 'AnonymizedFaker' or 'RegexGenerator'.
Use 'get_config()' to see the current transformer assignments.
```

### PR DESCRIPTION
**Pull request by pvk-developer**
Resolves #546

### CODE REVIEW

File: rdt/hyper_transformer.py [hunk 1]
```diff
@@ -792,6 +794,52 @@ def fit_transform(self, data):
...
+        if unknown_columns:
+            raise Error(
+                f"Unknown column name {unknown_columns}. Use 'get_config()' to see a "
+                'list of valid column names.'
+            )
+
+        transformers = []
+        for column_name in column_names:
+            transformer = self._transformers_tree.get(column_name, {}).get('transformer')
+            if not isinstance(transformer, get_generator_transformers()):
```
Comment (amontanez24) [inline rdt/hyper_transformer.py:828]: is this method really necessary? Can't we just do 
```python
if transformer.IS_GENERATOR
```

File: rdt/hyper_transformer.py [hunk 2]
```diff
@@ -792,6 +794,52 @@ def fit_transform(self, data):
...
+                Number of rows to be created. Must be an integer greater than 0.
+            column_names (list):
+                List of column names to be created.
+
+        Returns:
+            pandas.DataFrame:
+                A data frame with the newly generated columns of the size ``num_rows``.
+        """
+        if not self._fitted or self._modified_config:
+            raise NotFittedError(self._NOT_FIT_MESSAGE)
```
Comment (amontanez24) [inline rdt/hyper_transformer.py:813]: minor: let's just pull this chunk out into a `_validate_fitted` method that raises the error. This is used in 3 places now

**Review by amontanez24 — APPROVED** [review]
A couple of minor comments but this LGTM!

File: rdt/hyper_transformer.py [hunk 3]
```diff
@@ -792,10 +796,54 @@ def fit_transform(self, data):
...
+                A data frame with the newly generated columns of the size ``num_rows``.
+        """
+        self._validate_fitted()
+
+        if not isinstance(num_rows, int) or num_rows <= 0:
+            raise Error("Parameter 'num_rows' must be an integer greater than 0.")
+
+        unknown_columns = self._subset(column_names, self._input_columns, not_in=True)
+        if unknown_columns:
+            raise Error(
```
Comment (katxiao) [inline rdt/hyper_transformer.py:821]: This is out of scope for this PR, but I just wanted to leave a comment - `Error` seems a little generic. Maybe something like `HyperTransformerError` would be more informative?

**Review by katxiao — APPROVED** [review]
LGTM!

**Pull request merged**